### ISSUE TITLE
10_deploy_rook.sh exits with error: no matching resources found

### ISSUE THREAD

**Issue by mcornea** [openshift-metalkube/dev-scripts#351]
```shell
[centos@provisionhost-0 dev-scripts]$ ./10_deploy_rook.sh | awk '{ print strftime("%Y-%m-%d %H:%M:%S |"), $0; fflush(); }'
+ source logging.sh
+++ dirname ./10_deploy_rook.sh
++ LOGDIR=./logs
++ '[' '!' -d ./logs ']'
+++ basename ./10_deploy_rook.sh .sh
+++ date +%F-%H%M%S
++ LOGFILE=./logs/10_deploy_rook-2019-04-13-175424.log
++ echo 'Logging to ./logs/10_deploy_rook-2019-04-13-175424.log'
++ exec
2019-04-13 17:54:24 | Logging to ./logs/10_deploy_rook-2019-04-13-175424.log
+++ tee ./logs/10_deploy_rook-2019-04-13-175424.log
2019-04-13 17:54:24 | + source common.sh
2019-04-13 17:54:24 | ++++ dirname common.sh
2019-04-13 17:54:24 | +++ cd .
2019-04-13 17:54:24 | +++ pwd
2019-04-13 17:54:24 | ++ SCRIPTDIR=/home/centos/dev-scripts
2019-04-13 17:54:24 | +++ whoami
2019-04-13 17:54:24 | ++ USER=centos
2019-04-13 17:54:24 | ++ '[' -z '' ']'
2019-04-13 17:54:24 | ++ '[' -f /home/centos/dev-scripts/config_centos.sh ']'
2019-04-13 17:54:24 | ++ echo 'Using CONFIG /home/centos/dev-scripts/config_centos.sh'
2019-04-13 17:54:24 | Using CONFIG /home/centos/dev-scripts/config_centos.sh
2019-04-13 17:54:24 | ++ CONFIG=/home/centos/dev-scripts/config_centos.sh
2019-04-13 17:54:24 | ++ source /home/centos/dev-scripts/config_centos.sh
2019-04-13 17:54:24 | +++ set +x
2019-04-13 17:54:24 | +++ BOOTSTRAP_SSH_READY=2500
2019-04-13 17:54:24 | +++ NODES_PLATFORM=baremetal
2019-04-13 17:54:24 | +++ INT_IF=eth0
2019-04-13 17:54:24 | +++ PRO_IF=eth1
2019-04-13 17:54:24 | +++ ROOT_DISK=/dev/vda
2019-04-13 17:54:24 | +++ NODES_FILE=/home/centos/instackenv.json
2019-04-13 17:54:24 | +++ MANAGE_BR_BRIDGE=n
2019-04-13 17:54:24 | ++ ADDN_DNS=
2019-04-13 17:54:24 | ++ EXT_IF=
2019-04-13 17:54:24 | ++ PRO_IF=eth1
2019-04-13 17:54:24 | ++ MANAGE_BR_BRIDGE=n
2019-04-13 17:54:24 | ++ MANAGE_PRO_BRIDGE=y
2019-04-13 17:54:24 | ++ MANAGE_INT_BRIDGE=y
2019-04-13 17:54:24 | ++ INT_IF=eth0
2019-04-13 17:54:24 | ++ ROOT_DISK=/dev/vda
2019-04-13 17:54:24 | ++ KAFKA_NAMESPACE=strimzi
2019-04-13 17:54:24 | ++ KAFKA_CLUSTERNAME=strimzi
2019-04-13 17:54:24 | ++ KAFKA_PVC_SIZE=10
2019-04-13 17:54:24 | ++ KAFKA_PRODUCER_TIMER=100
2019-04-13 17:54:24 | ++ KAFKA_PRODUCER_TOPIC=strimzi-topic
2019-04-13 17:54:24 | ++ FILESYSTEM=/
2019-04-13 17:54:24 | ++ WORKING_DIR=/opt/dev-scripts
2019-04-13 17:54:24 | ++ NODES_FILE=/home/centos/instackenv.json
2019-04-13 17:54:24 | ++ NODES_PLATFORM=baremetal
2019-04-13 17:54:24 | ++ MASTER_NODES_FILE=ocp/master_nodes.json
2019-04-13 17:54:24 | ++ export RHCOS_IMAGE_URL=https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/
2019-04-13 17:54:24 | ++ RHCOS_IMAGE_URL=https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa/
2019-04-13 17:54:24 | +++ jq -r '.builds[1]'
2019-04-13 17:54:24 | +++ curl https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa//builds.json
2019-04-13 17:54:24 |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2019-04-13 17:54:24 |                                  Dload  Upload   Total   Spent    Left  Speed
100   737  100   737    0     0   1032      0 --:--:-- --:--:-- --:--:--  1033
2019-04-13 17:54:25 | ++ export RHCOS_LATEST=410.8.20190412.2
2019-04-13 17:54:25 | ++ RHCOS_LATEST=410.8.20190412.2
2019-04-13 17:54:25 | ++ export RHCOS_IMAGE_VERSION=410.8.20190412.2
2019-04-13 17:54:25 | ++ RHCOS_IMAGE_VERSION=410.8.20190412.2
2019-04-13 17:54:25 | +++ curl https://releases-rhcos.svc.ci.openshift.org/storage/releases/ootpa//410.8.20190412.2/meta.json
2019-04-13 17:54:25 | +++ jq -r .images.openstack.path
2019-04-13 17:54:25 |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2019-04-13 17:54:25 |                                  Dload  Upload   Total   Spent    Left  Speed
100  4778  100  4778    0     0  10485      0 --:--:-- --:--:-- --:--:-- 10478
2019-04-13 17:54:25 | ++ export RHCOS_IMAGE_FILENAME_OPENSTACK_GZ=rhcos-410.8.20190412.2-openstack.qcow2
2019-04-13 17:54:25 | ++ RHCOS_IMAGE_FILENAME_OPENSTACK_GZ=rhcos-410.8.20190412.2-openstack.qcow2
2019-04-13 17:54:25 | +++ echo rhcos-410.8.20190412.2-openstack.qcow2
2019-04-13 17:54:25 | +++ sed -e 's/-openstack.*//'
2019-04-13 17:54:25 | ++ export RHCOS_IMAGE_NAME=rhcos-410.8.20190412.2
2019-04-13 17:54:25 | ++ RHCOS_IMAGE_NAME=rhcos-410.8.20190412.2
2019-04-13 17:54:25 | ++ export RHCOS_IMAGE_FILENAME_OPENSTACK=rhcos-410.8.20190412.2-openstack.qcow2
2019-04-13 17:54:25 | ++ RHCOS_IMAGE_FILENAME_OPENSTACK=rhcos-410.8.20190412.2-openstack.qcow2
2019-04-13 17:54:25 | ++ export RHCOS_IMAGE_FILENAME_COMPRESSED=rhcos-410.8.20190412.2-compressed.qcow2
2019-04-13 17:54:25 | ++ RHCOS_IMAGE_FILENAME_COMPRESSED=rhcos-410.8.20190412.2-compressed.qcow2
2019-04-13 17:54:25 | ++ export RHCOS_IMAGE_FILENAME_LATEST=rhcos-ootpa-latest.qcow2
2019-04-13 17:54:25 | ++ RHCOS_IMAGE_FILENAME_LATEST=rhcos-ootpa-latest.qcow2
2019-04-13 17:54:25 | ++ export IRONIC_IMAGE=quay.io/metalkube/metalkube-ironic
2019-04-13 17:54:25 | ++ IRONIC_IMAGE=quay.io/metalkube/metalkube-ironic
2019-04-13 17:54:25 | ++ export IRONIC_INSPECTOR_IMAGE=quay.io/metalkube/metalkube-ironic-inspector
2019-04-13 17:54:25 | ++ IRONIC_INSPECTOR_IMAGE=quay.io/metalkube/metalkube-ironic-inspector
2019-04-13 17:54:25 | ++ export IRONIC_DATA_DIR=/opt/dev-scripts/ironic
2019-04-13 17:54:25 | ++ IRONIC_DATA_DIR=/opt/dev-scripts/ironic
2019-04-13 17:54:25 | ++ export KUBECONFIG=/home/centos/dev-scripts/ocp/auth/kubeconfig
2019-04-13 17:54:25 | ++ KUBECONFIG=/home/centos/dev-scripts/ocp/auth/kubeconfig
2019-04-13 17:54:25 | ++ export 'SSH=ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5'
2019-04-13 17:54:25 | ++ SSH='ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5'
2019-04-13 17:54:25 | ++ export LIBVIRT_DEFAULT_URI=qemu:///system
2019-04-13 17:54:25 | ++ LIBVIRT_DEFAULT_URI=qemu:///system
2019-04-13 17:54:25 | ++ '[' centos '!=' root -a /run/user/1000 == /run/user/0 ']'
2019-04-13 17:54:25 | ++ sudo -n uptime
2019-04-13 17:54:25 | +++ awk -F= '/^ID=/ { print $2 }' /etc/os-release
2019-04-13 17:54:25 | +++ tr -d '"'
2019-04-13 17:54:25 | ++ [[ ! centos =~ ^(centos|rhel)$ ]]
2019-04-13 17:54:25 | +++ awk -F= '/^VERSION_ID=/ { print $2 }' /etc/os-release
2019-04-13 17:54:25 | +++ tr -d '"'
2019-04-13 17:54:25 | +++ cut -f1 -d.
2019-04-13 17:54:25 | ++ [[ 7 -ne 7 ]]
2019-04-13 17:54:25 | +++ df / --output=fstype
2019-04-13 17:54:25 | +++ grep -v Type
2019-04-13 17:54:25 | ++ FSTYPE=xfs
2019-04-13 17:54:25 | ++ case ${FSTYPE} in
2019-04-13 17:54:25 | +++ xfs_info /
2019-04-13 17:54:25 | +++ grep -q ftype=1
2019-04-13 17:54:25 | ++ [[ -n '' ]]
2019-04-13 17:54:25 | ++ '[' 477 = 0 ']'
2019-04-13 17:54:25 | ++ '[' '!' -d /opt/dev-scripts ']'
2019-04-13 17:54:25 | + figlet 'Deploying rook'
2019-04-13 17:54:25 | + lolcat
2019-04-13 17:54:25 |  ____             _             _                               _
2019-04-13 17:54:25 | |  _ \  ___ _ __ | | ___  _   _(_)_ __   __ _   _ __ ___   ___ | | __
2019-04-13 17:54:25 | | | | |/ _ \ '_ \| |/ _ \| | | | | '_ \ / _` | | '__/ _ \ / _ \| |/ /
2019-04-13 17:54:25 | | |_| |  __/ |_) | | (_) | |_| | | | | | (_| | | | | (_) | (_) |   <
2019-04-13 17:54:25 | |____/ \___| .__/|_|\___/ \__, |_|_| |_|\__, | |_|  \___/ \___/|_|\_\
2019-04-13 17:54:25 |            |_|            |___/         |___/
2019-04-13 17:54:25 | ++ go env
2019-04-13 17:54:25 | + eval 'GOARCH="amd64"
2019-04-13 17:54:25 | GOBIN=""
2019-04-13 17:54:25 | GOCACHE="/home/centos/.cache/go-build"
2019-04-13 17:54:25 | GOEXE=""
2019-04-13 17:54:25 | GOFLAGS=""
2019-04-13 17:54:25 | GOHOSTARCH="amd64"
2019-04-13 17:54:25 | GOHOSTOS="linux"
2019-04-13 17:54:25 | GOOS="linux"
2019-04-13 17:54:25 | GOPATH="/home/centos/go"
2019-04-13 17:54:25 | GOPROXY=""
2019-04-13 17:54:25 | GORACE=""
2019-04-13 17:54:25 | GOROOT="/usr/lib/golang"
2019-04-13 17:54:25 | GOTMPDIR=""
2019-04-13 17:54:25 | GOTOOLDIR="/usr/lib/golang/pkg/tool/linux_amd64"
2019-04-13 17:54:25 | GCCGO="gccgo"
2019-04-13 17:54:25 | CC="gcc"
2019-04-13 17:54:25 | CXX="g++"
2019-04-13 17:54:25 | CGO_ENABLED="1"
2019-04-13 17:54:25 | GOMOD=""
2019-04-13 17:54:25 | CGO_CFLAGS="-g -O2"
2019-04-13 17:54:25 | CGO_CPPFLAGS=""
2019-04-13 17:54:25 | CGO_CXXFLAGS="-g -O2"
2019-04-13 17:54:25 | CGO_FFLAGS="-g -O2"
2019-04-13 17:54:25 | CGO_LDFLAGS="-g -O2"
2019-04-13 17:54:25 | PKG_CONFIG="pkg-config"
2019-04-13 17:54:25 | GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build160591485=/tmp/go-build -gno-record-gcc-switches"'
2019-04-13 17:54:25 | ++ GOARCH=amd64
2019-04-13 17:54:25 | ++ GOBIN=
2019-04-13 17:54:25 | ++ GOCACHE=/home/centos/.cache/go-build
2019-04-13 17:54:25 | ++ GOEXE=
2019-04-13 17:54:25 | ++ GOFLAGS=
2019-04-13 17:54:25 | ++ GOHOSTARCH=amd64
2019-04-13 17:54:25 | ++ GOHOSTOS=linux
2019-04-13 17:54:25 | ++ GOOS=linux
2019-04-13 17:54:25 | ++ GOPATH=/home/centos/go
2019-04-13 17:54:25 | ++ GOPROXY=
2019-04-13 17:54:25 | ++ GORACE=
2019-04-13 17:54:25 | ++ GOROOT=/usr/lib/golang
2019-04-13 17:54:25 | ++ GOTMPDIR=
2019-04-13 17:54:25 | ++ GOTOOLDIR=/usr/lib/golang/pkg/tool/linux_amd64
2019-04-13 17:54:25 | ++ GCCGO=gccgo
2019-04-13 17:54:25 | ++ CC=gcc
2019-04-13 17:54:25 | ++ CXX=g++
2019-04-13 17:54:25 | ++ CGO_ENABLED=1
2019-04-13 17:54:25 | ++ GOMOD=
2019-04-13 17:54:25 | ++ CGO_CFLAGS='-g -O2'
2019-04-13 17:54:25 | ++ CGO_CPPFLAGS=
2019-04-13 17:54:25 | ++ CGO_CXXFLAGS='-g -O2'
2019-04-13 17:54:25 | ++ CGO_FFLAGS='-g -O2'
2019-04-13 17:54:25 | ++ CGO_LDFLAGS='-g -O2'
2019-04-13 17:54:25 | ++ PKG_CONFIG=pkg-config
2019-04-13 17:54:25 | ++ GOGCCFLAGS='-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build160591485=/tmp/go-build -gno-record-gcc-switches'
2019-04-13 17:54:25 | + export ROOKPATH=/home/centos/go/src/github.com/rook/rook
2019-04-13 17:54:25 | + ROOKPATH=/home/centos/go/src/github.com/rook/rook
2019-04-13 17:54:25 | + cd /home/centos/go/src/github.com/rook/rook/cluster/examples/kubernetes/ceph
2019-04-13 17:54:25 | + oc create -f common.yaml
2019-04-13 17:54:26 | namespace/rook-ceph created
2019-04-13 17:54:26 | customresourcedefinition.apiextensions.k8s.io/cephclusters.ceph.rook.io created
2019-04-13 17:54:26 | customresourcedefinition.apiextensions.k8s.io/cephfilesystems.ceph.rook.io created
2019-04-13 17:54:26 | customresourcedefinition.apiextensions.k8s.io/cephnfses.ceph.rook.io created
2019-04-13 17:54:26 | customresourcedefinition.apiextensions.k8s.io/cephobjectstores.ceph.rook.io created
2019-04-13 17:54:26 | customresourcedefinition.apiextensions.k8s.io/cephobjectstoreusers.ceph.rook.io created
2019-04-13 17:54:26 | customresourcedefinition.apiextensions.k8s.io/cephblockpools.ceph.rook.io created
2019-04-13 17:54:26 | customresourcedefinition.apiextensions.k8s.io/volumes.rook.io created
2019-04-13 17:54:26 | clusterrole.rbac.authorization.k8s.io/rook-ceph-cluster-mgmt created
2019-04-13 17:54:26 | role.rbac.authorization.k8s.io/rook-ceph-system created
2019-04-13 17:54:26 | clusterrole.rbac.authorization.k8s.io/rook-ceph-global created
2019-04-13 17:54:26 | clusterrole.rbac.authorization.k8s.io/rook-ceph-mgr-cluster created
2019-04-13 17:54:26 | serviceaccount/rook-ceph-system created
2019-04-13 17:54:26 | rolebinding.rbac.authorization.k8s.io/rook-ceph-system created
2019-04-13 17:54:26 | clusterrolebinding.rbac.authorization.k8s.io/rook-ceph-global created
2019-04-13 17:54:26 | serviceaccount/rook-ceph-osd created
2019-04-13 17:54:26 | serviceaccount/rook-ceph-mgr created
2019-04-13 17:54:26 | role.rbac.authorization.k8s.io/rook-ceph-osd created
2019-04-13 17:54:26 | clusterrole.rbac.authorization.k8s.io/rook-ceph-mgr-system created
2019-04-13 17:54:26 | role.rbac.authorization.k8s.io/rook-ceph-mgr created
2019-04-13 17:54:26 | rolebinding.rbac.authorization.k8s.io/rook-ceph-cluster-mgmt created
2019-04-13 17:54:26 | rolebinding.rbac.authorization.k8s.io/rook-ceph-osd created
2019-04-13 17:54:26 | rolebinding.rbac.authorization.k8s.io/rook-ceph-mgr created
2019-04-13 17:54:26 | rolebinding.rbac.authorization.k8s.io/rook-ceph-mgr-system created
2019-04-13 17:54:26 | clusterrolebinding.rbac.authorization.k8s.io/rook-ceph-mgr-cluster created
2019-04-13 17:54:26 | + sed '/FLEXVOLUME_DIR_PATH/!b;n;c\          value: "\/etc/kubernetes\/kubelet-plugins\/volume\/exec"' operator-openshift.yaml
2019-04-13 17:54:26 | + sed -i 's/# - name: FLEXVOLUME_DIR_PATH/- name: FLEXVOLUME_DIR_PATH/' operator-openshift-modified.yaml
2019-04-13 17:54:26 | + oc create -f operator-openshift-modified.yaml
2019-04-13 17:54:26 | securitycontextconstraints.security.openshift.io/rook-ceph created
2019-04-13 17:54:26 | deployment.apps/rook-ceph-operator created
2019-04-13 17:54:26 | + oc wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
2019-04-13 17:54:26 | error: no matching resources found
```

**Comment by mcornea** [comment]
If I wait for a few seconds then it returns correctly:

```shell
[centos@provisionhost-0 dev-scripts]$ oc --config ocp/auth/kubeconfig wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
error: no matching resources found
[centos@provisionhost-0 dev-scripts]$ oc --config ocp/auth/kubeconfig wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
error: no matching resources found
[centos@provisionhost-0 dev-scripts]$ oc --config ocp/auth/kubeconfig wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
error: no matching resources found
[centos@provisionhost-0 dev-scripts]$ oc --config ocp/auth/kubeconfig wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
error: no matching resources found
[centos@provisionhost-0 dev-scripts]$ oc --config ocp/auth/kubeconfig wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s; echo $?
error: no matching resources found
1
[centos@provisionhost-0 dev-scripts]$ oc --config ocp/auth/kubeconfig wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
error: no matching resources found
[centos@provisionhost-0 dev-scripts]$ oc --config ocp/auth/kubeconfig wait --for condition=ready pod -l app=rook-ceph-operator -n rook-ceph --timeout=120s
pod/rook-ceph-operator-654b5c69b5-gbpjk condition met
```

**Comment by karmab** [comment]
write, we should add a sleep there... weirdly enough, it works fine on virtual setup

**Comment by umangachapagain** [comment]
Fixed by /pull/353 ??

**Comment by hardys** [comment]
#353 landed, and I just tested locally, all seems to work so closing this as fixed (although long term we need a better solution than sleeping in scripts!)
